pkg/auth/manager/publicshares/publicshares.go

```diff
@@ -122,6 +132,9 @@ func (m *manager) Authenticate(ctx context.Context, token, secret string) (*user
 	case publicShareResponse.Status.Code == rpcv1beta1.Code_CODE_NOT_FOUND:
 		return nil, nil, errtypes.NotFound(publicShareResponse.Status.Message)
 	case publicShareResponse.Status.Code == rpcv1beta1.Code_CODE_PERMISSION_DENIED:
```

Comment:
Why do we expect `PERMISSION DENIED` here? Shouldn't we expect `TOO MANY REQUESTS`?

Reply:
At this point, we've allowed the request to go through the brute force protection (`bfp.Verify` method) and we're trying the get the share with the provided password. We get the permission denied if the provided password is wrong.